scargo/docker_base/esp32.Dockerfile

```diff
@@ -0,0 +1,114 @@
+###
+# @copyright Copyright (C) 2022 Spyrosoft Solutions. All rights reserved.
+# THIS FILE WAS GENERATED AUTOMATICALLY. DO NOT CHANGE IT.
+###
+
+ARG DOCKER_IMAGE_ROOT=ubuntu:20.04
+FROM ${DOCKER_IMAGE_ROOT} as base
+
+ENV DEBIAN_FRONTEND noninteractive
+
+RUN apt update --fix-missing && \
+    apt -y install python3.8 python3.8-venv python3-pip \
+    binutils git wget \
+    scons build-essential pkg-config \ 
+    unzip graphviz nano vim && \
+    apt -y install sudo && \
+    update-alternatives --install /usr/bin/python python /usr/bin/python3 1 && \
+    update-alternatives --install /usr/bin/pip pip /usr/bin/pip3 1
+
+FROM base AS cpp
+
+RUN apt -y install cppcheck bzr lib32z1 \
+    clang clang-format clang-tidy valgrind \
+    gcovr doxygen curl libcurl4-openssl-dev \
+    libcmocka0 libcmocka-dev plantuml
+
+FROM cpp AS esp32
+RUN  apt -y install flex bison gperf python3-setuptools ninja-build ccache libffi-dev libssl-dev dfu-util libusb-1.0-0
+
+ARG ESPRESSIF_IDF_TAG="v4.4.3"
+ARG ESPRESSIF_IDF_VERSION="${ESPRESSIF_IDF_TAG}"
+ARG ESPRESSIF_IDF_PATH="/opt/esp-idf"
+ARG ESPRESSIF_IDF_REPO_URL="https://github.com/espressif/esp-idf.git"
+ARG ESPRESSIF_TOOLS_PATH="/root/.espressif"
+ENV  IDF_TOOLS_PATH ${ESPRESSIF_IDF_PATH}
+ENV  IDF_PATH ${ESPRESSIF_IDF_PATH}
+
+WORKDIR /opt
+
+RUN umask 0002 && git clone --single-branch --branch ${ESPRESSIF_IDF_VERSION} --recursive ${ESPRESSIF_IDF_REPO_URL} ${ESPRESSIF_IDF_PATH}
+RUN umask 0002 && ${ESPRESSIF_IDF_PATH}/install.sh
+
+RUN curl -sL https://aka.ms/InstallAzureCLIDeb | sudo bash && \
+    apt-get update && \
+    apt-get install -y ca-certificates curl apt-transport-https lsb-release gnupg && \
+    curl -sL https://packages.microsoft.com/keys/microsoft.asc | \
+    gpg --dearmor | \
+    tee /etc/apt/trusted.gpg.d/microsoft.gpg > /dev/null && \
+    AZ_REPO=$(lsb_release -cs) && \
+    echo "deb [arch=amd64] https://packages.microsoft.com/repos/azure-cli/ $AZ_REPO main" | \
+    tee /etc/apt/sources.list.d/azure-cli.list && \ 
+    apt-get update && \
+    apt-get install azure-cli 
+
+ENV LC_ALL=C
+
+RUN umask 0002 && mkdir /opt/esp/ && echo '#!/usr/bin/env bash\n\
+\n\
+set -e\n\
+\n\
+. $IDF_PATH/export.sh\n\
+\n\
+exec "$@"' >  /opt/esp/entrypoint.sh && chmod 775 /opt/esp/entrypoint.sh
+    
+# Install QEMU
+ARG QEMU_VER=esp-develop-20220919
+ARG QEMU_DIST=qemu-${QEMU_VER}.tar.bz2
+ARG QEMU_SHA256=f6565d3f0d1e463a63a7f81aec94cce62df662bd42fc7606de4b4418ed55f870
+RUN : \
+  && wget --no-verbose https://github.com/espressif/qemu/releases/download/${QEMU_VER}/${QEMU_DIST} \
+  && echo "${QEMU_SHA256} *${QEMU_DIST}" | sha256sum --check --strict - \
+  && tar -xf ${QEMU_DIST} -C /opt \
+  && rm ${QEMU_DIST} \
+  && :
+ENV PATH=/opt/qemu/bin:${PATH}
+WORKDIR /opt
+
+FROM esp32 AS custom_espproj
```

Comment:
I cannot see the possibility here to :
1) add a user needed libraries or tool without excluding docker from update
2) updating the docker with the new scargo version if changes arrive (e.g. the optimization which we plan in the near future)